include/zephyr/bluetooth/buf.h

```diff
@@ -48,15 +50,60 @@ enum bt_buf_type {
 	BT_BUF_ISO_OUT = BIT(4),
 	/** Incoming ISO data */
 	BT_BUF_ISO_IN = BIT(5),
-	/** H:4 data */
-	BT_BUF_H4 = BIT(6),
 };
 
-/** @brief This is a base type for bt_buf user data. */
-struct bt_buf_data {
-	uint8_t type;
+/** Direction of HCI packets. Only used for mapping H:4 to BT_BUF_* values. */
+enum bt_buf_dir {
+	BT_BUF_IN,
+	BT_BUF_OUT,
```

Comment:
Consider adding documentation for each value here, if only to be able to reach 100% coverage on https://docs.zephyrproject.org/api-coverage/latest/zephyr/bluetooth/buf.h.gcov.html